frontend/src/components/message.tsx

```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+import React, { useMemo } from 'react';
+import styles from './message.module.css';
+import UserIcon from '../icons/user.svg';
+import BotIcon from '../icons/cpu.svg';
+
+export enum Role {
+  User = 'User',
+  Bot = 'Bot',
+}
+
+export interface Message {
+  role: Role;
+  content: string;
+  time: string;
+}
+
+export interface MessageProps {
+  message: Message;
+}
+
+export interface MessageStyle {
+  icon: string;
+  class: string;
+}
+
+const roleStyleMap: Record<Role, MessageStyle> = {
+  [Role.User]: {
+    icon: UserIcon,
+    class: styles.user,
+  },
+  [Role.Bot]: {
+    icon: BotIcon,
+    class: styles.bot,
+  },
+};
+
+export const MessageComponent = ({ message }: MessageProps) => {
+  const { content, role } = message;
+
+  const { class: roleClass, icon } = useMemo(() => roleStyleMap[role], [role]);
+
+  return (
+    <div className={classNames(styles.container, roleClass)}>
+      <img src={icon} />
+      <p>{content}</p>
+    </div>
+  );
+};
```

Comment:
So if I understand this correctly, a message to the backend looks like this:

- MessageComponent
   - Takes MessageProps
     - Contains [content, role, time] which are the message content, who said it and the time it was said
   - returns html/JSX which contains the message and roleClass

And the RoleClass converts the Role into the csv and identifier for the bot or the user?

Reply:
not quite, the `roleClass` is just a CSS class that determines the position of the message in the chat. 

Currently, `server.ts` is the only file that will interact with the backend, and atm is just taking a `message: string` as the only argument. We can change this to whatever is needed in the wiring up step

Reply:
Cool cool, in that case for my first connection PR I'll just adjust the `server.ts` to take the response I sent it from the backend 👍